### DISCUSSION TITLE
are --private firejail files accessible outside sandbox in any other way than firejail --get/--ls/--put/--cat?

### DISCUSSION THREAD

**Discussion by danielkrajnik** [netblue30/firejail#4822] [Q&A] [answered]
does firejail in `--private` mode physically saves files to disk? is there any directory in a regular file system where files stored by firejail in `--private` mode are accessible to the host? when firejail uses namespaces to create a sandbox, which lives for the duration of a program are files saved by that program in sandbox accessible for other users/root?

**Comment by rusty-snake** [comment, marked as answer]
> is there any directory in a regular file system where files stored by firejail in --private mode are accessible to the host?

Depends what a "regular file system" is for you.

https://github.com/netblue30/firejail/discussions/4746#discussioncomment-1767519

> > IIRC, downloaded files in a --private instance aren't directly accessible by your filesystem so I can't just go there and mv it for retrieval.
> 
> root can: `sudo ls /proc/<child pid>/root/$HOME`
> note: `<child pid>` is the pid of the process running in the sandbox not firejail. Get it from `firejail --tree`.

> accessible for root

yes (root -> kernel -> anything)

> does firejail in --private mode physically saves files to disk?

No, it is a `tmpfs`.

If your swap partition  is on a physically disk ...

> when firejail uses namespaces to create a sandbox, which lives for the duration of a program are files saved by that program in sandbox accessible for other users/root?

You can `nsenter` a namespace.

**Reply by danielkrajnik** [reply]
thank you very much, that clarifies a lot. 

> No, it is a `tmpfs`.

So if a program in `--private` mode tried to save a file to disk larger than Ram+Swap, it would cause errors?

**Reply by rusty-snake** [reply]
I'm not sure at which point (`tmpfs`s can have size limits IKD firejail sets one) but it will not work.
